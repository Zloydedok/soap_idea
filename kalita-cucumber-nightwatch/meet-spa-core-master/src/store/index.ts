import Vue from 'vue';
import Vuex from 'vuex';

import {DELETE_ITEM, EDIT_ITEM, SET_ITEMS} from './mutation-types';

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    items: [],
    loading: true
  },
  mutations: {
    [SET_ITEMS](state, payload) {
      state.items = payload;
      state.loading = false;
    },
    [DELETE_ITEM](state, index: number) {
      state.items.splice(index, 1);
    },
    [EDIT_ITEM](state, payload) {
      Vue.set(state.items, payload.index, payload.item);
    }
  }
});
