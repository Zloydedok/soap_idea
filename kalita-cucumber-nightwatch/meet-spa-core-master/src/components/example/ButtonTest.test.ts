import {mount} from '@vue/test-utils';
import ButtonTest from './ButtonTest.vue';
import store from '../../store/index';

describe('ButtonTest', () => {
  const wrapper = mount(ButtonTest, {store});

  it('should render correct contents', () => {
    expect(wrapper.vm.$el.querySelector('button').textContent).toMatchSnapshot();
  });

  it('should press the button', () => {
    wrapper.vm.pressButton();
    expect(wrapper.vm.pressed).toEqual(true);
  });
});
