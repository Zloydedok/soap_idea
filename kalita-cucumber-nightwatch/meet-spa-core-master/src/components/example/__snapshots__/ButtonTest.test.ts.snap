// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonTest should render correct contents 1`] = `
"Test Button
  "
`;
